File: 2389. Longest Subsequence With Limited Sum/Solution.py
from typing import List


class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        result = []
        nums.sort()
        for query in queries:
            result.append(self.answerQuery(nums, query))

        return result

    def answerQuery(self, nums: List[int], query: int) -> int:
        sum = 0
        result = 0
        for num in nums:
            sum += num
            if (sum > query):
                break
            else:
                result = result+1
        return result

File: 2389. Longest Subsequence With Limited Sum/Solution.java
import java.util.Arrays;

class Solution {

    public int[] answerQueries(int[] nums, int[] queries) {
        int[] result = new int[queries.length];
        Arrays.sort(nums);
        for (int i = 0; i < result.length; i++) {
            result[i] = maxLength(nums, queries[i]);
        }
        return result;
    }

    public int maxLength(int[] nums, int query) {
        int sum = 0;
        int result = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            if (sum <= query) {
                result++;
            } else {
                break;
            }
        }
        return result;
    }
}